airbyte_cdk/test/standard_tests/_job_runner.py

```diff
@@ -112,12 +112,18 @@ def run_test_job(
     # This is a bit of a hack because the source needs the catalog early.
     # Because it *also* can fail, we have to redundantly wrap it in a try/except block.
 
+    expect_exception = False
+    if test_scenario and test_scenario.expect_exception:
+        # If the test scenario expects an exception, we need to set the
+        # `expect_exception` flag to True.
+        expect_exception = True
+
     result: entrypoint_wrapper.EntrypointOutput = entrypoint_wrapper._run_command(  # noqa: SLF001  # Non-public API
         source=connector_obj,  # type: ignore [arg-type]
         args=args,
-        expecting_exception=test_scenario.expect_exception,
+        expecting_exception=False if not test_scenario else expect_exception,
```

Comment:
I forgot to update this after adding the dedicated variable. Fixes:

```suggestion
        expecting_exception=expect_exception,
```